File: aula-01/exemplos/exemplo3.py
class ContaBancaria:
    def __init__(self, titular, saldo=0):
        self.titular = titular
        self.saldo = saldo

    def depositar(self, valor):
        if valor > 0:
            self.saldo += valor

    def sacar(self, valor):
        if 0 < valor <= self.saldo:
            self.saldo -= valor
        else:
            print("Saldo insuficiente!")

    def exibir_saldo(self):
        print(f"Titular: {self.titular} | Saldo: R$ {self.saldo:.2f}")


contas = []

while True:
    nome = input("Nome do titular (ou 'sair'): ")
    if nome.lower() == "sair":
        break

    conta = ContaBancaria(nome)
    contas.append(conta)

for conta in contas:
    conta.depositar(1000)
    conta.sacar(250)
    conta.exibir_saldo()

File: aula-01/exemplos/Exemplo3.java
package org.example;

import java.util.ArrayList;
import java.util.Scanner;

public class Exemplo3 {

    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);
        ArrayList<ContaBancaria> contas = new ArrayList<>();

        while (true) {
            System.out.print("Nome do titular (ou 'sair'): ");
            String nome = scanner.nextLine();

            if (nome.equalsIgnoreCase("sair")) {
                break;
            }

            ContaBancaria conta = new ContaBancaria();
            conta.titular = nome;
            contas.add(conta);
        }

        for (ContaBancaria conta : contas) {
            conta.depositar(1000);
            conta.sacar(250);
            conta.exibirSaldo();
        }

    }
}
